backend/de.metas.contracts/src/test/java/de/metas/contracts/bpartner/process/C_BPartner_Location_updateFromPreviousAddressTest.java

```diff
@@ -0,0 +1,139 @@
+		final I_C_BPartner_Location bpartnerLocation2 = newInstance(I_C_BPartner_Location.class);
+		bpartnerLocation2.setC_BPartner_ID(bpartner.getC_BPartner_ID());
+		bpartnerLocation2.setC_Location_ID(2);
+		bpartnerLocation2.setPrevious_ID(bpartnerLocation1.getC_BPartner_Location_ID());
+		bpartnerLocation2.setValidFrom(TimeUtil.addWeeks(now, -10));
+		bpartnerLocation2.setIsActive(true);
+		bpartnerLocation2.setIsBillTo(true);
+		bpartnerLocation2.setIsShipTo(true);
+		bpartnerLocation2.setIsBillToDefault(true);
+		bpartnerLocation2.setIsShipToDefault(true);
+		save(bpartnerLocation2);
+		current_cBpartnerLocationId = bpartnerLocation2.getC_BPartner_Location_ID();
+
+		// replacing location
+		final I_C_BPartner_Location bpartnerLocation3 = newInstance(I_C_BPartner_Location.class);
+		bpartnerLocation3.setC_BPartner_ID(bpartner.getC_BPartner_ID());
+		bpartnerLocation3.setC_Location_ID(3);
+		bpartnerLocation3.setPrevious_ID(bpartnerLocation2.getC_BPartner_Location_ID());
+		bpartnerLocation3.setValidFrom(TimeUtil.addWeeks(now, -1));
+		save(bpartnerLocation3);
+		replacing_cBpartnerLocationId = bpartnerLocation3.getC_BPartner_Location_ID();
+
+		// replacing location
+		final I_C_BPartner_Location bpartnerLocation4 = newInstance(I_C_BPartner_Location.class);
+		bpartnerLocation4.setC_BPartner_ID(bpartner.getC_BPartner_ID());
+		bpartnerLocation4.setC_Location_ID(4);
+		bpartnerLocation4.setPrevious_ID(bpartnerLocation2.getC_BPartner_Location_ID());
+		bpartnerLocation4.setValidFrom(TimeUtil.addWeeks(now, 1));
+		bpartnerLocation4.setIsBillTo(true);
+		save(bpartnerLocation4);
+		notYetActive_cBpartnerLocationId = bpartnerLocation4.getC_BPartner_Location_ID();
+	}
+
+	@Test
+	public void testSuccessiveUpdates() throws Exception
+	{
+		final C_BPartner_Location_UpdateFromPreviousAddress c_bPartner_location_updateFromPreviousAddress = new C_BPartner_Location_UpdateFromPreviousAddress()
+		{
+			protected void logDeactivation(final Map<Integer, Integer> oldLocToNewLoc, final Integer locationId)
+			{
+				System.out.println("Replacing " + locationId + " with " + oldLocToNewLoc.get(locationId));
+			}
+		};
+
+		c_bPartner_location_updateFromPreviousAddress.doIt();
+		final I_C_BPartner_Location oldLoc = load(current_cBpartnerLocationId, I_C_BPartner_Location.class);
+		assertThat(oldLoc.isActive()).isFalse();
+		assertThat(oldLoc.isBillTo()).isTrue();
+		assertThat(oldLoc.isShipTo()).isTrue();
+		assertThat(oldLoc.isBillToDefault()).isFalse();
+		assertThat(oldLoc.isShipToDefault()).isFalse();
+
+		final I_C_BPartner_Location newLoc = load(replacing_cBpartnerLocationId, I_C_BPartner_Location.class);
+		assertThat(newLoc.isActive()).isTrue();
+		assertThat(newLoc.isBillTo()).isTrue();
+		assertThat(newLoc.isShipTo()).isTrue();
+		assertThat(newLoc.isBillToDefault()).isTrue();
+		assertThat(newLoc.isShipToDefault()).isTrue();
+
+		final I_C_BPartner_Location notYetActiveLoc = load(notYetActive_cBpartnerLocationId, I_C_BPartner_Location.class);
```

Comment:
note the best naming - can cause FUD in hte future with `assertThat(notYetActiveLoc.isActive()).isTrue();`

Reply:
made sense when writing it, does not when reading it again. Renamed.